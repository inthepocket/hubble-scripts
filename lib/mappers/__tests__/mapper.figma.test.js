const getFixtures = require('./__fixtures__/mapper.figma');
const getMapper = require('..');

let FIXTURES = {};
let mappers;

describe('figma mapper test suite', () => {
  beforeAll(() => {
    mappers = getMapper(false); // false means no sketchfile
    FIXTURES = getFixtures();
  })

  describe('-> #formatColors:', () => {
    it('Given figma formatted array of JSON colors, should transform to a nicely parsed array', () => {
        expect(FIXTURES.COLORS.map(mappers.colors)).toMatchSnapshot();
    })
  })

  describe('-> #formatTextStyles:', () => {
    it('Given figma formatted array of JSON textstyles, should transform to a nicely parsed array', () => {
      expect(FIXTURES.TEXT_STYLES.map(mappers.textStyles)).toMatchSnapshot();
    })
  })

  describe('-> #formatGradients:', () => {
    it('Given figma formatted array of JSON gradients, shoudl transform to a nicely parsed array', () => {
      expect(FIXTURES.GRADIENTS.map(mappers.gradients)).toMatchSnapshot();
    })
  })

  describe('-> #formatShadows', () => {
    it('Given figma formatted array of JSON shadows, should transform to a nicely parsed array', () => {
      expect(FIXTURES.SHADOWS.map(mappers.shadows)).toMatchSnapshot();
    })
  })

  describe('-> #formatBorders:', () => {
    it('Given figma formatted array of JSON borders, should transform to a nicely parsed array', () => {
      expect(FIXTURES.BORDERS.map(mappers.borders)).toMatchSnapshot();
    })
  })
})
  
 
