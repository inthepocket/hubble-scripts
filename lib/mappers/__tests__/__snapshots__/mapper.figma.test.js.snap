// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`figma mapper test suite -> #formatBorders: Given figma formatted array of JSON borders, should transform to a nicely parsed array 1`] = `
Array [
  Object {
    "color": Object {
      "alpha": 1,
      "blue": 150,
      "green": 150,
      "hex": "ff9696",
      "hsl": Array [
        0,
        100,
        79,
      ],
      "name": "geraldine",
      "red": 255,
      "rgb": Array [
        255,
        150,
        150,
      ],
      "rgba": Array [
        255,
        150,
        150,
        1,
      ],
      "token": "color",
    },
    "id": "inside",
    "radius": 3,
    "token": "border",
    "type": "INSIDE",
    "width": 1,
  },
  Object {
    "color": Object {
      "alpha": 1,
      "blue": 150,
      "green": 150,
      "hex": "ff9696",
      "hsl": Array [
        0,
        100,
        79,
      ],
      "name": "geraldine",
      "red": 255,
      "rgb": Array [
        255,
        150,
        150,
      ],
      "rgba": Array [
        255,
        150,
        150,
        1,
      ],
      "token": "color",
    },
    "id": "outside",
    "radius": 12.5,
    "token": "border",
    "type": "OUTSIDE",
    "width": 1,
  },
  Object {
    "gradient": Object {
      "from": Array [
        1,
        1,
      ],
      "name": null,
      "stops": Array [
        Object {
          "color": Object {
            "alpha": 1,
            "blue": 144,
            "green": 182,
            "hex": "97b690",
            "hsl": Array [
              109,
              21,
              64,
            ],
            "name": "norway",
            "red": 151,
            "rgb": Array [
              151,
              182,
              144,
            ],
            "rgba": Array [
              151,
              182,
              144,
              1,
            ],
            "token": "color",
          },
          "position": 0,
        },
        Object {
          "color": Object {
            "alpha": 1,
            "blue": 88,
            "green": 145,
            "hex": "559158",
            "hsl": Array [
              123,
              26,
              45,
            ],
            "name": "hippieGreen",
            "red": 85,
            "rgb": Array [
              85,
              145,
              88,
            ],
            "rgba": Array [
              85,
              145,
              88,
              1,
            ],
            "token": "color",
          },
          "position": 0.5038507580757141,
        },
        Object {
          "color": Object {
            "alpha": 1,
            "blue": 62,
            "green": 125,
            "hex": "227d3e",
            "hsl": Array [
              138,
              57,
              31,
            ],
            "name": "salem",
            "red": 34,
            "rgb": Array [
              34,
              125,
              62,
            ],
            "rgba": Array [
              34,
              125,
              62,
              1,
            ],
            "token": "color",
          },
          "position": 1,
        },
      ],
      "to": Array [
        1.5,
        0.9999999999999999,
      ],
      "token": "gradient",
      "type": "GRADIENT_LINEAR",
    },
    "id": "gradient-with-corners",
    "radius": Array [
      8,
      12.5,
      12.5,
      12.5,
    ],
    "token": "border",
    "type": "OUTSIDE",
    "width": 1,
  },
  Object {
    "color": Object {
      "alpha": 1,
      "blue": 150,
      "green": 150,
      "hex": "ff9696",
      "hsl": Array [
        0,
        100,
        79,
      ],
      "name": "geraldine",
      "red": 255,
      "rgb": Array [
        255,
        150,
        150,
      ],
      "rgba": Array [
        255,
        150,
        150,
        1,
      ],
      "token": "color",
    },
    "id": "center",
    "radius": 0,
    "token": "border",
    "type": "CENTER",
    "width": 1,
  },
]
`;

exports[`figma mapper test suite -> #formatColors: Given figma formatted array of JSON colors, should transform to a nicely parsed array 1`] = `
Array [
  Object {
    "alpha": 1,
    "blue": 230,
    "green": 194,
    "hex": "5fc2e6",
    "hsl": Array [
      196,
      73,
      64,
    ],
    "name": "seagull",
    "red": 95,
    "rgb": Array [
      95,
      194,
      230,
    ],
    "rgba": Array [
      95,
      194,
      230,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 209,
    "green": 193,
    "hex": "5ec1d1",
    "hsl": Array [
      188,
      56,
      59,
    ],
    "name": "viking",
    "red": 94,
    "rgb": Array [
      94,
      193,
      209,
    ],
    "rgba": Array [
      94,
      193,
      209,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 214,
    "green": 200,
    "hex": "77c8d6",
    "hsl": Array [
      189,
      54,
      65,
    ],
    "name": "viking",
    "red": 119,
    "rgb": Array [
      119,
      200,
      214,
    ],
    "rgba": Array [
      119,
      200,
      214,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 219,
    "green": 207,
    "hex": "8dcfdb",
    "hsl": Array [
      189,
      52,
      71,
    ],
    "name": "halfBaked",
    "red": 141,
    "rgb": Array [
      141,
      207,
      219,
    ],
    "rgba": Array [
      141,
      207,
      219,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 224,
    "green": 214,
    "hex": "a0d6e0",
    "hsl": Array [
      189,
      51,
      75,
    ],
    "name": "powderBlue",
    "red": 160,
    "rgb": Array [
      160,
      214,
      224,
    ],
    "rgba": Array [
      160,
      214,
      224,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 230,
    "green": 221,
    "hex": "b3dde6",
    "hsl": Array [
      191,
      50,
      80,
    ],
    "name": "powderBlue",
    "red": 179,
    "rgb": Array [
      179,
      221,
      230,
    ],
    "rgba": Array [
      179,
      221,
      230,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 240,
    "green": 235,
    "hex": "d4ebf0",
    "hsl": Array [
      191,
      48,
      89,
    ],
    "name": "pattensBlue",
    "red": 212,
    "rgb": Array [
      212,
      235,
      240,
    ],
    "rgba": Array [
      212,
      235,
      240,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 250,
    "green": 243,
    "hex": "e5f3fa",
    "hsl": Array [
      200,
      68,
      94,
    ],
    "name": "solitude",
    "red": 229,
    "rgb": Array [
      229,
      243,
      250,
    ],
    "rgba": Array [
      229,
      243,
      250,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 253,
    "green": 249,
    "hex": "f2f9fd",
    "hsl": Array [
      202,
      73,
      97,
    ],
    "name": "aliceBlue",
    "red": 242,
    "rgb": Array [
      242,
      249,
      253,
    ],
    "rgba": Array [
      242,
      249,
      253,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 62,
    "green": 125,
    "hex": "227d3e",
    "hsl": Array [
      138,
      57,
      31,
    ],
    "name": "salem",
    "red": 34,
    "rgb": Array [
      34,
      125,
      62,
    ],
    "rgba": Array [
      34,
      125,
      62,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 94,
    "green": 94,
    "hex": "f55e5e",
    "hsl": Array [
      0,
      88,
      66,
    ],
    "name": "carnation",
    "red": 245,
    "rgb": Array [
      245,
      94,
      94,
    ],
    "rgba": Array [
      245,
      94,
      94,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 74,
    "green": 135,
    "hex": "3e874a",
    "hsl": Array [
      130,
      37,
      39,
    ],
    "name": "seaGreen",
    "red": 62,
    "rgb": Array [
      62,
      135,
      74,
    ],
    "rgba": Array [
      62,
      135,
      74,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 88,
    "green": 145,
    "hex": "559158",
    "hsl": Array [
      123,
      26,
      45,
    ],
    "name": "hippieGreen",
    "red": 85,
    "rgb": Array [
      85,
      145,
      88,
    ],
    "rgba": Array [
      85,
      145,
      88,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 106,
    "green": 157,
    "hex": "6b9d6a",
    "hsl": Array [
      119,
      21,
      52,
    ],
    "name": "aquaForest",
    "red": 107,
    "rgb": Array [
      107,
      157,
      106,
    ],
    "rgba": Array [
      107,
      157,
      106,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 124,
    "green": 169,
    "hex": "81a97c",
    "hsl": Array [
      113,
      21,
      57,
    ],
    "name": "amulet",
    "red": 129,
    "rgb": Array [
      129,
      169,
      124,
    ],
    "rgba": Array [
      129,
      169,
      124,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 144,
    "green": 182,
    "hex": "97b690",
    "hsl": Array [
      109,
      21,
      64,
    ],
    "name": "norway",
    "red": 151,
    "rgb": Array [
      151,
      182,
      144,
    ],
    "rgba": Array [
      151,
      182,
      144,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 186,
    "green": 210,
    "hex": "c0d2ba",
    "hsl": Array [
      105,
      21,
      78,
    ],
    "name": "paleLeaf",
    "red": 192,
    "rgb": Array [
      192,
      210,
      186,
    ],
    "rgba": Array [
      192,
      210,
      186,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 208,
    "green": 224,
    "hex": "d4e0d0",
    "hsl": Array [
      105,
      21,
      85,
    ],
    "name": "tasman",
    "red": 212,
    "rgb": Array [
      212,
      224,
      208,
    ],
    "rgba": Array [
      212,
      224,
      208,
      1,
    ],
    "token": "color",
  },
  Object {
    "alpha": 1,
    "blue": 231,
    "green": 239,
    "hex": "e9efe7",
    "hsl": Array [
      105,
      20,
      92,
    ],
    "name": "grayNurse",
    "red": 233,
    "rgb": Array [
      233,
      239,
      231,
    ],
    "rgba": Array [
      233,
      239,
      231,
      1,
    ],
    "token": "color",
  },
]
`;

exports[`figma mapper test suite -> #formatGradients: Given figma formatted array of JSON gradients, shoudl transform to a nicely parsed array 1`] = `
Array [
  Object {
    "from": Array [
      0.5038167964273101,
      1.0000000430625262,
    ],
    "name": "linear",
    "stops": Array [
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 144,
          "green": 182,
          "hex": "97b690",
          "hsl": Array [
            109,
            21,
            64,
          ],
          "name": "norway",
          "red": 151,
          "rgb": Array [
            151,
            182,
            144,
          ],
          "rgba": Array [
            151,
            182,
            144,
            1,
          ],
          "token": "color",
        },
        "position": 0,
      },
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 88,
          "green": 145,
          "hex": "559158",
          "hsl": Array [
            123,
            26,
            45,
          ],
          "name": "hippieGreen",
          "red": 85,
          "rgb": Array [
            85,
            145,
            88,
          ],
          "rgba": Array [
            85,
            145,
            88,
            1,
          ],
          "token": "color",
        },
        "position": 0.5038507580757141,
      },
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 62,
          "green": 125,
          "hex": "227d3e",
          "hsl": Array [
            138,
            57,
            31,
          ],
          "name": "salem",
          "red": 34,
          "rgb": Array [
            34,
            125,
            62,
          ],
          "rgba": Array [
            34,
            125,
            62,
            1,
          ],
          "token": "color",
        },
        "position": 1,
      },
    ],
    "to": Array [
      1.0038167964273075,
      0.9999999674926328,
    ],
    "token": "gradient",
    "type": "FRAME",
  },
  Object {
    "from": Array [
      0.4999999999999999,
      1.0000000000000002,
    ],
    "name": "radial",
    "stops": Array [
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 144,
          "green": 182,
          "hex": "97b690",
          "hsl": Array [
            109,
            21,
            64,
          ],
          "name": "norway",
          "red": 151,
          "rgb": Array [
            151,
            182,
            144,
          ],
          "rgba": Array [
            151,
            182,
            144,
            1,
          ],
          "token": "color",
        },
        "position": 0,
      },
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 62,
          "green": 125,
          "hex": "227d3e",
          "hsl": Array [
            138,
            57,
            31,
          ],
          "name": "salem",
          "red": 34,
          "rgb": Array [
            34,
            125,
            62,
          ],
          "rgba": Array [
            34,
            125,
            62,
            1,
          ],
          "token": "color",
        },
        "position": 1,
      },
    ],
    "to": Array [
      1.7726315302406337,
      1.0000000000000002,
    ],
    "token": "gradient",
    "type": "FRAME",
  },
  Object {
    "from": Array [
      0.5,
      0.5,
    ],
    "name": "angular",
    "stops": Array [
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 144,
          "green": 182,
          "hex": "97b690",
          "hsl": Array [
            109,
            21,
            64,
          ],
          "name": "norway",
          "red": 151,
          "rgb": Array [
            151,
            182,
            144,
          ],
          "rgba": Array [
            151,
            182,
            144,
            1,
          ],
          "token": "color",
        },
        "position": 0.4336458444595337,
      },
      Object {
        "color": Object {
          "alpha": 1,
          "blue": 62,
          "green": 125,
          "hex": "227d3e",
          "hsl": Array [
            138,
            57,
            31,
          ],
          "name": "salem",
          "red": 34,
          "rgb": Array [
            34,
            125,
            62,
          ],
          "rgba": Array [
            34,
            125,
            62,
            1,
          ],
          "token": "color",
        },
        "position": 1,
      },
    ],
    "to": Array [
      0.5,
      1,
    ],
    "token": "gradient",
    "type": "FRAME",
  },
  Object {
    "from": null,
    "name": "gradient-with-corners",
    "stops": null,
    "to": null,
    "token": "gradient",
    "type": "FRAME",
  },
]
`;

exports[`figma mapper test suite -> #formatShadows Given figma formatted array of JSON shadows, should transform to a nicely parsed array 1`] = `
Array [
  Object {
    "blur": undefined,
    "color": Object {
      "alpha": 0.5,
      "blue": 0,
      "green": 0,
      "hex": "000000",
      "hsl": Array [
        0,
        0,
        0,
      ],
      "name": "black",
      "red": 0,
      "rgb": Array [
        0,
        0,
        0,
      ],
      "rgba": Array [
        0,
        0,
        0,
        0.5,
      ],
      "token": "color",
    },
    "id": "simple",
    "spread": null,
    "token": "shadow",
    "type": "drop-shadow",
    "x": 0,
    "y": 2,
  },
  Object {
    "blur": undefined,
    "color": Object {
      "alpha": 0.5,
      "blue": 0,
      "green": 0,
      "hex": "000000",
      "hsl": Array [
        0,
        0,
        0,
      ],
      "name": "black",
      "red": 0,
      "rgb": Array [
        0,
        0,
        0,
      ],
      "rgba": Array [
        0,
        0,
        0,
        0.5,
      ],
      "token": "color",
    },
    "id": "simple",
    "spread": null,
    "token": "shadow",
    "type": "inner-shadow",
    "x": 0,
    "y": 1,
  },
]
`;

exports[`figma mapper test suite -> #formatTextStyles: Given figma formatted array of JSON textstyles, should transform to a nicely parsed array 1`] = `
Array [
  Object {
    "borderBottom": null,
    "family": "FiraSans-Book",
    "id": "paragraph",
    "kerning": null,
    "lineHeight": 21.1,
    "paragraphSpacing": null,
    "size": 18,
    "token": "text-style",
    "weight": "book",
  },
  Object {
    "borderBottom": null,
    "family": "Helvetica-Bold",
    "id": "quote",
    "kerning": 1,
    "lineHeight": 16.4,
    "paragraphSpacing": null,
    "size": 14,
    "token": "text-style",
    "weight": "bold",
  },
  Object {
    "borderBottom": null,
    "family": "FiraSans-Regular",
    "id": "heading2",
    "kerning": null,
    "lineHeight": 42.2,
    "paragraphSpacing": null,
    "size": 36,
    "token": "text-style",
    "weight": "regular",
  },
  Object {
    "borderBottom": null,
    "family": "FiraSans-Bold",
    "id": "heading1",
    "kerning": null,
    "lineHeight": 57.6,
    "paragraphSpacing": null,
    "size": 48,
    "token": "text-style",
    "weight": "bold",
  },
]
`;
